src/renderer/qss/js/qssFocusManager.js

```diff
@@ -0,0 +1,331 @@
+        },
+
+        listeners: {
+            "onArrowUpPressed.impl": {
+                funcName: "gpii.qss.qssFocusManager.onArrowUpPressed",
+                args: ["{that}"]
+            },
+            "onArrowDownPressed.impl": {
+                funcName: "gpii.qss.qssFocusManager.onArrowDownPressed",
+                args: ["{that}"]
+            },
+            "onArrowLeftPressed.impl": {
+                funcName: "gpii.qss.qssFocusManager.onArrowLeftPressed",
+                args: ["{that}"]
+            },
+            "onArrowRightPressed.impl": {
+                funcName: "gpii.qss.qssFocusManager.onArrowRightPressed",
+                args: ["{that}"]
+            }
+        },
+
+        invokers: {
+            getFocusInfo: {
+                funcName: "gpii.qss.qssFocusManager.getFocusInfo",
+                args: ["{that}.container", "{that}.options.styles"]
+            },
+            getFocusGroupsInfo: {
+                funcName: "gpii.qss.qssFocusManager.getFocusGroupsInfo",
+                args: ["{that}", "{that}.container"]
+            }
+        }
+    });
+
+    /**
+     * Retrieves the tab index of a given DOM or jQuery element as a number.
+     * @param {HTMLElement | jQuery} element - A simple DOM element or wrapped in a jQuery
+     * object.
+     * @return {Number} The tab index of the given element or `undefined` if this property
+     * is not specified.
+     */
+    gpii.qss.qssFocusManager.getTabIndex = function (element) {
+        var tabIndex = jQuery(element).attr("tabindex");
+        if (tabIndex) {
+            return parseInt(tabIndex, 10);
+        }
+    };
+
+    /**
+     * Returns information about the focusable elements in the QSS as well as the index of
+     * the currently focused element. "Focusable element" in terms of the QSS means an element
+     * with a "fl-focusable" class which has a specified `tabindex` greater than or equal to 0.
+     * The focusable elements are returned sorted ascendingly by their tab index.
+     * @param {jQuery} container - The jQuery element representing the container in which this
+     * focus manager handles focus.
+     * @param {Object} styles - A styles object containing various classes related to focusing
+     * of elements
+     * @return {Object} Information about the focusable elements.
+     */
+    gpii.qss.qssFocusManager.getFocusInfo = function (container, styles) {
+        var focusableElements =
+            container
+                .find("." + styles.focusable + ":visible")
+                .filter(function () {
+                    var tabindex = gpii.qss.qssFocusManager.getTabIndex(this);
+                    return tabindex >= 0;
+                })
+                .sort(function (left, right) {
+                    var leftTabIndex = gpii.qss.qssFocusManager.getTabIndex(left),
+                        rightTabIndex = gpii.qss.qssFocusManager.getTabIndex(right);
+                    return leftTabIndex - rightTabIndex;
```

Comment:
No, the function passed as an argument to the `sort` operation should return a number which is less than 0 if the left argument is smaller than the right, greater than 0 if the left argument is bigger than the right and 0 otherwise.